File: LeetCode/contains-duplicate.cpp
// https://leetcode.com/problems/contains-duplicate/

class Solution
{
public:
	bool containsDuplicate(vector<int> &nums)
	{
		unordered_map<int, int> m;
		unordered_map<int, int>::iterator itr;

		for (auto num : nums)
		{
			itr = m.find(num);
			if (itr != m.end())
			{
				itr->second++;
				if (itr->second > 1)
				{
					return true;
				}
			}
			else
			{
				m.insert({num, 1});
			}
		}

		return false;
	}
};

File: LeetCode/Java/containsduplicate/ContainsDuplicate.java
package containsduplicate;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
	public static void main(String[] args) {
		Solution sol = new Solution();
		boolean result = sol.containsDuplicate(new int[]{1, 2, 3, 1});
		System.out.println(result);
	}
}

// https://leetcode.com/problems/contains-duplicate
//IMPORTANT!! Submit Code Region Begin(Do not remove this line)
class Solution {
	public boolean containsDuplicate(int[] nums) {
		Set<Integer> set = new HashSet<>();

		for (int num : nums) {
			if (set.contains(num)) {
				return true;
			}
			set.add(num);
		}

		return false;
	}
}
//IMPORTANT!! Submit Code Region End(Do not remove this line)
